backend/src/test/java/com/hdi/backend/CheckupActions/CheckupActionServiceTest.java

```diff
@@ -0,0 +1,164 @@
+
+        // THEN
+        verify(mockCheckupActionRepository).findAll();
+        assertNotNull(actualActions);
+        assertEquals(expectedActions, actualActions);
+    }
+
+    @Test
+    void shouldReturnEmptyList_whenRepositoryIsEmpty() {
+        // GIVEN
+        when(mockCheckupActionRepository.findAll()).thenReturn(List.of());
+
+        // WHEN
+        List<CheckupAction> actualActions = checkupActionService.getAllActions();
+
+        // THEN
+        verify(mockCheckupActionRepository).findAll();
+        assertNotNull(actualActions);
+        assertTrue(actualActions.isEmpty());
+    }
+
+    @Test
+    void shouldReturnAllCheckupActions_whenRepositoryContainsMultipleActions() {
+        // GIVEN
+        CheckupAction action1 = CheckupAction.builder()
+                .id("1")
+                .title("test1")
+                .build();
+        CheckupAction action2 = CheckupAction.builder()
+                .id("2")
+                .title("test2")
+                .build();
+        List<CheckupAction> expectedActions = List.of(action1, action2);
+
+        when(mockCheckupActionRepository.findAll()).thenReturn(expectedActions);
+
+        // WHEN
+        List<CheckupAction> actualActions = checkupActionService.getAllActions();
+
+        // THEN
+        verify(mockCheckupActionRepository).findAll();
+        assertNotNull(actualActions);
+        assertEquals(expectedActions, actualActions);
+    }
+
+
+    @Test
+    void shouldReturnCheckupActionById_whenActionExists() {
+        //GIVEN
+        String id = "1";
+        CheckupAction action1 = CheckupAction.builder()
+                .id(id)
+                .title("test")
+                .build();
+        when(mockCheckupActionRepository.findById(id)).thenReturn(Optional.of(action1));
+        //WHEN
+        CheckupAction actual = checkupActionService.getActionById(id);
+        //THEN
+        verify(mockCheckupActionRepository).findById(id);
+        assertEquals(action1, actual);
+    }
+
+    @Test
+    void shouldThrowException_whenActionNotFound(){
+        //GIVEN
+        String notExistentId = "9999";
+        when(mockCheckupActionRepository.findById(notExistentId)).thenReturn(Optional.empty());
+
+        //WHEN & THEN
+        assertThrows(NoSuchElementException.class, () -> checkupActionService.getActionById(notExistentId));
+        verify(mockCheckupActionRepository).findById(notExistentId);
+    }
+
+    @Test
+    void shouldSaveAndReturnCheckupActionWithGeneratedId() {
+        // GIVEN
+        CheckupAction actionWithoutId = CheckupAction.builder()
+                .title("test")
+                .build();
+
+        CheckupAction savedAction = CheckupAction.builder()
+                .id("generatedMongoId123")
+                .title("test")
+                .build();
+
+        when(mockCheckupActionRepository.save(any(CheckupAction.class))).thenReturn(savedAction);
```

Comment:
Ich würde das `any()`wenn möglichst vermeiden, es lässt offen welcher Parameter übergeben werden kann
Wenn du jetzt im Service den title veränderst wird das in diesem Test nicht auffallen.